packages/peregrine/lib/talons/WishlistPage/__tests__/useCreateWishlist.spec.js

```diff
@@ -1,57 +1,121 @@
 import React from 'react';
 
-import createTestInstance from '../../../util/createTestInstance';
+import { MockedProvider } from '@apollo/client/testing';
+import { renderHook, act } from '@testing-library/react-hooks';
+
 import { useCreateWishlist } from '../useCreateWishlist';
+import defaultOperations from '../createWishlist.gql';
+import wishlistPageOperations from '../wishlistPage.gql';
 
-const Component = props => {
-    const talonProps = useCreateWishlist(props);
+jest.mock('@magento/peregrine/lib/context/app', () => {
+    const state = {};
+    const api = { actions: { setPageLoading: jest.fn() } };
+    const useAppContext = jest.fn(() => [state, api]);
 
-    return <i talonProps={talonProps} />;
-};
+    return { useAppContext };
+});
 
-const getTalonProps = props => {
-    const tree = createTestInstance(<Component {...props} />);
-    const { root } = tree;
-    const { talonProps } = root.findByType('i').props;
+const createWishlistVariables = {
+    name: 'Test WishList',
+    visibility: 'PUBLIC'
+};
 
-    const update = newProps => {
-        tree.update(<Component {...{ ...props, ...newProps }} />);
+let createWishlistMutationCalled = false;
+let getCustomerWishlistsQueryCalled = false;
+
+const getCustomerWishlistsMock = {
+    request: {
+        query: wishlistPageOperations.getCustomerWishlistQuery
+    },
+    loading: false,
+    result: () => {
+        getCustomerWishlistsQueryCalled = true;
+        return {
+            data: {
+                customer: {
+                    id: 'customerId',
+                    wishlists: [
+                        {
+                            id: 42,
+                            items_count: 0,
+                            name: 'Test WishList',
+                            visibility: 'PUBLIC',
+                            sharing_code: 'code'
+                        }
+                    ]
+                }
+            }
+        };
+    }
+};
 
-        return root.findByType('i').props.talonProps;
-    };
+const createWishlistMock = {
```

Comment:
How do you test the loading state of this mutation? I'm trying to do this in my PR but I'm having trouble...

Reply:
Nvm, found it :D Just don't `await` the act'ed click.